src/hotspot/share/gc/serial/markSweep.cpp

```diff
@@ -163,12 +163,14 @@ void MarkSweep::mark_object(oop obj) {
     _string_dedup_requests->add(obj);
   }
 
+  // Do the transform while we still have the header intact,
+  // which might include important class information.
+  ContinuationGCSupport::transform_stack_chunk(obj);
```

Comment:
Not sure if position of  calling ContinuationGCSupport::transform_stack_chunk will have side effect. Had better guard with option...